js/packages/teams-ai/src/Application.ts

```diff
@@ -471,6 +527,23 @@ export class Application<TState extends TurnState = DefaultTurnState> {
                 const { storage, turnStateManager } = this._options;
                 const state = await turnStateManager!.loadState(storage, context);
 
+                // Sign the user in
+                if (this._authentication && (await this._startSignIn?.(context))) {
+                    const response = await this._authentication.signUserIn(context, state);
+
+                    if (response.status == 'pending') {
+                        // Save turn state
+                        // - This lets the bot keep track of authentication state
+                        await turnStateManager!.saveState(storage, context, state);
+                        return false;
+                    }
+
+                    // Activities for which auth cannot be initiated should be ignored
```

Comment:
This isn't ignoring though, this is throwing the app.

Reply:
It's ignoring if `response.cause == 'invalidActivity'`

Reply:
The code says if it does not equal (`!=`) invalidActivity

Reply:
yeah so the code says if `response.cause != 'invalidActivity'` then throw an error. This implies that if it's an invalid activity then you ignore. The reason for the comment is that the logic is a bit convoluted. 

I can rewrite it to look like
```ts
if (response.status == 'error') {
    if (response.cause == 'invalidActivity') {
        // do nothing.
    } else {
        throw error;
    }
}
```

I prefer the style I implemented it in because it's concise. Which do you prefer?

Reply:
Okay. I guess the comment + the way the code is done confused me. I do like the conciseness but in situations like this the comment should preclude any ambiguity. How about: 

// Invalid activities will be ignored. Activities that cannot initiate auth will throw an error.

Reply:
The two sentences mean two opposite things. How about:

// Invalid activities will be ignored. Any other errors will be re thrown.